File: Catalog/Examples/Editor/OpenVideo/FromMultipleVideoClips/OpenVideoFromMultipleVideoClips.m
#import "OpenVideoFromMultipleVideoClips.h"
@import AVFoundation;
@import VideoEditorSDK;

@interface OpenVideoFromMultipleVideoClipsObjC () <PESDKVideoEditViewControllerDelegate>

@end

@implementation OpenVideoFromMultipleVideoClipsObjC

- (void)invokeExample {
  // Get the URLs for video files on the local file system. This could be any file within the app bundle
  // or also a file within the documents or temporary folders for example. This example will use
  // files from the app bundle.
  // highlight-setup
  NSURL *firstURL = [NSBundle.mainBundle URLForResource:@"Skater" withExtension:@"mp4"];
  NSURL *secondURL = [NSBundle.mainBundle URLForResource:@"test_video" withExtension:@"mp4"];

  // Create `AVAsset`s from above `NSURL`s.
  AVAsset *firstAsset = [AVAsset assetWithURL:firstURL];
  AVAsset *secondAsset = [AVAsset assetWithURL:secondURL];

  // Create a `PESDKVideo` from above assets.
  PESDKVideo *video = [[PESDKVideo alloc] initWithAssets:@[firstAsset, secondAsset]];
  // highlight-setup

  // Create a default `PESDKConfiguration`.
  PESDKConfiguration *configuration = [[PESDKConfiguration alloc] init];

  // Create a video editor and pass it the video and configuration. Make this class the delegate of it to handle export and cancelation.
  // highlight-instantiation
  PESDKVideoEditViewController *videoEditViewController = [[PESDKVideoEditViewController alloc] initWithVideoAsset:video configuration:configuration];
  videoEditViewController.delegate = self;
  // highlight-instantiation

  // Present the video editor.
  videoEditViewController.modalPresentationStyle = UIModalPresentationFullScreen;
  [self.presentingViewController presentViewController:videoEditViewController animated:YES completion:nil];
}

#pragma mark - PESDKVideoEditViewControllerDelegate

- (BOOL)videoEditViewControllerShouldStart:(PESDKVideoEditViewController * _Nonnull)videoEditViewController task:(PESDKVideoEditorTask * _Nonnull)task {
  // Implementing this method is optional. You can perform additional validation and interrupt the process by returning `NO`.
  return YES;
}

 - (void)videoEditViewControllerDidFinish:(PESDKVideoEditViewController * _Nonnull)videoEditViewController result:(PESDKVideoEditorResult * _Nonnull)result {
  // The user exported a new video successfully and the newly generated video is located at `result.output.url`. Dismissing the editor.
  [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

- (void)videoEditViewControllerDidFail:(PESDKVideoEditViewController * _Nonnull)videoEditViewController error:(PESDKVideoEditorError * _Nonnull)error {
  // There was an error generating the video.
  NSLog(@"%@", error.localizedDescription);
  // Dismissing the editor.
  [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

- (void)videoEditViewControllerDidCancel:(PESDKVideoEditViewController * _Nonnull)videoEditViewController {
  // The user tapped on the cancel button within the editor. Dismissing the editor.
  [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

@end

File: Catalog/Examples/Editor/OpenVideo/FromMultipleVideoClips/OpenVideoFromMultipleVideoClips.swift
import AVFoundation
import UIKit
import VideoEditorSDK

class OpenVideoFromMultipleVideoClipsSwift: Example, VideoEditViewControllerDelegate {
  override func invokeExample() {
    // Get the URLs for video files on the local file system. This could be any file within the app bundle
    // or also a file within the documents or temporary folders for example. This example will use
    // files from the app bundle.
    // highlight-setup
    let firstURL = Bundle.main.url(forResource: "Skater", withExtension: "mp4")!
    let secondURL = Bundle.main.url(forResource: "test_video", withExtension: "mp4")!

    // Create `AVAsset`s from above `URL`s.
    let firstAsset = AVAsset(url: firstURL)
    let secondAsset = AVAsset(url: secondURL)

    // Create a `Video` from above assets.
    let video = Video(assets: [firstAsset, secondAsset])
    // highlight-setup

    // Create a video editor and pass it the video. Make this class the delegate of it to handle export and cancelation.
    // highlight-instantiation
    let videoEditViewController = VideoEditViewController(videoAsset: video)
    videoEditViewController.delegate = self
    // highlight-instantiation

    // Present the video editor.
    videoEditViewController.modalPresentationStyle = .fullScreen
    presentingViewController?.present(videoEditViewController, animated: true, completion: nil)
  }

  // MARK: - VideoEditViewControllerDelegate

  func videoEditViewControllerShouldStart(_ videoEditViewController: VideoEditViewController, task: VideoEditorTask) -> Bool {
    // Implementing this method is optional. You can perform additional validation and interrupt the process by returning `false`.
    true
  }

  func videoEditViewControllerDidFinish(_ videoEditViewController: VideoEditViewController, result: VideoEditorResult) {
    // The user exported a new video successfully and the newly generated video is located at `result.output.url`. Dismissing the editor.
    presentingViewController?.dismiss(animated: true, completion: nil)
  }

  func videoEditViewControllerDidFail(_ videoEditViewController: VideoEditViewController, error: VideoEditorError) {
    // There was an error generating the video.
    print(error.localizedDescription)
    // Dismissing the editor.
    presentingViewController?.dismiss(animated: true, completion: nil)
  }

  func videoEditViewControllerDidCancel(_ videoEditViewController: VideoEditViewController) {
    // The user tapped on the cancel button within the editor. Dismissing the editor.
    presentingViewController?.dismiss(animated: true, completion: nil)
  }
}
